key.core/src/main/java/de/uka/ilkd/key/settings/ChoiceSettings.java

```diff
@@ -20,57 +12,56 @@
 import org.key_project.util.collection.ImmutableSLList;
 import org.key_project.util.collection.ImmutableSet;
 
-public class ChoiceSettings implements Settings, Cloneable {
+/**
+ *
+ */
+public class ChoiceSettings extends AbstractSettings {
+    private static final String KEY_DEFAULT_CHOICES = "[Choice]DefaultChoices";
 
-    private static final String DEFAULTCHOICES_KEY = "[Choice]DefaultChoices";
-    private LinkedList<SettingsListener> listenerList = new LinkedList<SettingsListener>();
+    private static final String PROP_CHOICE_DEFAULT = "category2Default";
+    private static final String PROP_CHOICE_CATEGORIES = "category2Choices";
     private HashMap<String, String> category2Default;
 
 
     /**
      * maps categories to a set of Strings(representing the choices which are options for this
      * category).
      */
-    private HashMap<String, Set<String>> category2Choices =
-        new LinkedHashMap<String, Set<String>>();
+    private Map<String, Set<String>> category2Choices = new LinkedHashMap<>();
 
 
     public ChoiceSettings() {
-        category2Default = new LinkedHashMap<String, String>();
+        category2Default = new LinkedHashMap<>();
     }
 
 
-    public ChoiceSettings(HashMap<String, String> category2Default) {
-        this.category2Default = category2Default;
+    public ChoiceSettings(Map<String, String> category2Default) {
+        this.category2Default = new HashMap<>(category2Default);
     }
 
 
-    public void setDefaultChoices(HashMap<String, String> category2Default) {
-        HashMap<String, String> category2Defaultold = this.category2Default;
-        this.category2Default = category2Default;
-        if (category2Defaultold != null && !category2Defaultold.equals(category2Default)) {
-            fireSettingsChanged();
-        }
+    public void setDefaultChoices(Map<String, String> category2Default) {
+        var old = this.category2Default;
+        this.category2Default = new HashMap<>(category2Default);
+        firePropertyChange(PROP_CHOICE_DEFAULT, old, this.category2Default);
     }
 
 
     /**
-     * returns a copy of the HashMap that maps categories to their choices.
+     * returns a copy of the HashMap that maps categories to
+     * their choices.
      */
-    @SuppressWarnings("unchecked")
-    public HashMap<String, Set<String>> getChoices() {
-        return (HashMap<String, Set<String>>) category2Choices.clone();
+    public Map<String, Set<String>> getChoices() {
+        return Collections.unmodifiableMap(category2Choices);
     }
 
-
     /**
      * returns a copy of the HashMap that maps categories to their currently selected choices.
```

Comment:
+1 for using unmodifiable instead of cloning, however, the comments are wrong now